datasets/cmu_hinglish_dog/cmu_hinglish_dog.py

```diff
@@ -108,70 +104,74 @@ def _split_generators(self, dl_manager):
-                p = Path(f)
-                os.rename(f, p.with_suffix(".json"))
-
         hi_en_dirs = {
             "train": os.path.join(data_dir_hi_en, "train"),
             "valid": os.path.join(data_dir_hi_en, "valid"),
             "test": os.path.join(data_dir_hi_en, "test"),
         }
 
-        en_dirs = {
-            "train": os.path.join(data_dir_en, "train"),
-            "valid": os.path.join(data_dir_en, "valid"),
-            "test": os.path.join(data_dir_en, "test"),
-        }
         return [
             datasets.SplitGenerator(
                 name=datasets.Split.TRAIN,
                 gen_kwargs={
                     "hi_en_dir": hi_en_dirs["train"],
-                    "en_dir": en_dirs["train"],
+                    "data_dir_en": data_dir_en,
                 },
             ),
             datasets.SplitGenerator(
                 name=datasets.Split.TEST,
-                gen_kwargs={"hi_en_dir": hi_en_dirs["test"], "en_dir": en_dirs["test"]},
+                gen_kwargs={
+                    "hi_en_dir": hi_en_dirs["test"],
+                    "data_dir_en": data_dir_en,
+                },
             ),
             datasets.SplitGenerator(
                 name=datasets.Split.VALIDATION,
                 gen_kwargs={
                     "hi_en_dir": hi_en_dirs["valid"],
-                    "en_dir": en_dirs["valid"],
+                    "data_dir_en": data_dir_en,
                 },
             ),
         ]
 
-    def _generate_examples(self, hi_en_dir, en_dir):
+    def _generate_examples(self, hi_en_dir, data_dir_en):
         """Yields examples."""
-        english_files = os.listdir(en_dir)
+        english_files_train = os.listdir(os.path.join(data_dir_en, "train"))
+        english_files_val = os.listdir(os.path.join(data_dir_en, "valid"))
+        english_files_test = os.listdir(os.path.join(data_dir_en, "test"))
+
         hinglish_files = os.listdir(hi_en_dir)
         key = 0
         for f in hinglish_files:
             en_file_path = f.split(".json")[0] + ".json"
-
-            if en_file_path in english_files:
-                en = json.load(open(os.path.join(en_dir, en_file_path)))
+            found = True
+            # Looks for the corresponding english file in all 3 splits
+            if en_file_path in english_files_train:
+                en = json.load(
+                    open(os.path.join(os.path.join(data_dir_en, "train"), en_file_path))
+                )
+            elif en_file_path in english_files_val:
+                en = json.load(
+                    open(os.path.join(os.path.join(data_dir_en, "valid"), en_file_path))
+                )
+            elif en_file_path in english_files_test:
+                en = json.load(
+                    open(os.path.join(os.path.join(data_dir_en, "test"), en_file_path))
+                )
+            else:
+                found = False
+            if found:
```

Comment:
I have modified the part of searching for the corresponding file in the english set. Earlier the English counterparts of a split where only being searched for in that specific split. I found that a lot of Hinglish files in ``val`` and ``test`` have english counterparts in ``train`` of English dir, so it didn't make sense to not use them.